tests/whenstmt/twhen.nim

```diff
@@ -0,0 +1,47 @@
+discard """
```

Comment:
neither of your tests demonstrates a difference between before PR and after PR

Reply:
It's not all that easy to replicate sadly, so I put in a test to ensure at least the basic behaviour is kept intact.

The failure is trying to build the vscode extension, in `jsffi`, the `{}` macro for instantiating JS objects fails, I spent over an hour building up various scenarios to create encapsualte the spec that's being violated, but besides resorting to a JS testcase I couldn't shrink it down. Since the only replication step is oddly JS specific, I wasn't inclined to comingle it with this and blindly shrinking it didn't demonstrate any real principle.

This is one of those things that would be better covered by one or all of:
* test suite that codifies the spec
* property based testing that uses a spec to generate code
* reworking the design to reduce the invalid states or guard against such errors

The above is all much better than a haphazard regression test. With all that said if you can figure out something better then I'm all ears.